src/index.js

```diff
@@ -79,9 +83,42 @@ export default class MarkersPlugin extends UICorePlugin {
       internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
     }
     internalMarker.onDestroy()
+    this._markers.splice(index, 1)
```

Comment:
I disagree object references will never change by themselves. If they did then javascript is broken :P. When a user creates a marker and passes it to `addMarker()` I am giving them the responsibility to keep hold of their reference if they want to delete it in the future.

I think it's pretty similar to the developer holding onto the index number they want to remove. The difference is with a number there's no absolute link but with a reference there is.

With the new  `getByIndex()` method if they did want to work with indexes though now they could (`removeMarker(getByIndex(12))`), but it would be a bit less efficient than holding onto the reference from the beginning.